discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -15,67 +15,75 @@
 use std::str::FromStr;
 
 use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
 use displaydoc::Display;
+use num_traits::FromPrimitive;
 use sqlx::{
     sqlite::{Sqlite, SqliteConnectOptions, SqlitePoolOptions},
     Pool,
+    QueryBuilder,
 };
 use thiserror::Error;
 use url::Url;
 use uuid::Uuid;
 use xayn_discovery_engine_ai::GenericError;
+use xayn_discovery_engine_providers::Market;
 
-use crate::document::{HistoricDocument, Id};
+use crate::document::{self, HistoricDocument};
 
-use super::Storage;
+use super::{
+    models::{ApiDocumentView, NewsResource, NewscatcherData},
+    FeedScope,
+    Storage,
+};
 
 #[derive(Error, Debug, Display)]
-pub enum Error {
-    /// Failed to initialize database: {0}
-    Init(#[source] sqlx::Error),
-
-    /// Failed to acquire connection: {0}
-    AcquireConnection(#[source] sqlx::Error),
-
+pub enum DatabaseError {
     /// Failed to migrate database: {0}
     Migration(#[from] sqlx::migrate::MigrateError),
 
-    /// Failed to fetch: {0}
-    Fetch(#[source] sqlx::Error),
+    /// Sql error: {0}
+    Sql(#[source] sqlx::Error),
 
     /// Failed to covert type: {0}
     TypeConversion(#[source] GenericError),
 }
 
+#[derive(Error, Debug, Display)]
+pub enum StorageError {
```

Comment:
@rustonaut 
I can see the benefit of splitting the errors for different traits. For example if we have one big error enum, we would have to match against all error variants, even though some may never occur.

```
pub enum StorageError {
    /// Database error: {0}
    Database(#[from] DatabaseError),
    History1(...),
    History2(...),
    Feed1(...),
    Feed2(...),
}

match storage.fetch_history() {
	Database =>
	History1 => 
	History2 => 
	Feed1 & Feed2 will never occur but have to handle them 
}
```

but yes, the question is whether we want to put any other checks into the storage implementation at all (like checking uuid, etc.)

Reply:
I don't think it makes ever sense to exhaustively match error variants, I mean we generally hardly match errors as most of them are bugs anyway.